plugin/environment.go

```diff
@@ -439,6 +447,11 @@ func (env *Environment) RunMultiPluginHook(hookRunnerFunc func(hooks Hooks) bool
 	})
 }
 
+// SetPrepackagedPlugins saves prepackaged plugins in the environment
+func (env *Environment) SetPrepackagedPlugins(plugins []*PrepackagedPlugin) {
+	env.prepackagedPlugins = plugins
```

Comment:
I think @hanzei's concern stems from the write to `prepackagedPlugins` being unsafe to do concurrently. I suspect all we need is a `RWMutex` within `Environment` to guard reads and writes against this variable.